### ISSUE TITLE
[environment]::Exit(0) breaks the interactive window

### ISSUE THREAD

**Issue by DataTraveler1** [ironmansoftware/powershell-universal#877]
### 1. Create a new PowerShell Form Project
![image](https://user-images.githubusercontent.com/537287/151045405-733a1f8c-ef06-4ac8-a4d3-b3618dbfd089.png)

### 2. Include the below code as part of your exit routine
```
[environment]::Exit(0)
```
### 3. Run the project and observe the error below
>
[ERROR] The background process reported an error with the following message: "The named pipe target process has ended.".
[ERROR] + CategoryInfo          : ResourceUnavailable: (:) [], ParentContainsErrorRecordException
[ERROR] + FullyQualifiedErrorId : System.Management.Automation.Remoting.PSRemotingDataStructureException
PS C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE> 
>
### 4. Subsequent attempts to run the project will end immediately with the below result
>
The program 'PowerShellFormProject1.ps1: PowerShell Script' has exited with code 0 (0x0).
>

## Expected behavior
```
[environment]::Exit(0) should end the running application without breaking the interactive window
```

## Actual behavior
```
The interactive window breaks which requires a restart of VS to resolve.
```

## Possible Mitigations
n/a

## Version

Visual Studio 2022.02
Windows 10 21H2

**Comment by DataTraveler1** [comment]
Correction: I was referring to the Output tab not the PowerShell Interactive Window

**Comment by adamdriscoll** [comment]
This also happens in VS Code. I think we should provide a way to restart the PS process without having to restart VS. Exiting the application is something we can try to guard against but I'd rather PS behave as expected but we can recover from it.

**Comment by DataTraveler1** [comment]
@adamdriscoll Thank you for looking into this and your guidance.

**Background**

This issue happened because I ported my code from a similar application and I left that part in to see what happens. It turns out that it is not needed for exiting gracefully on a PowerShell Pro Tools windows forms project. Thus, I have since commented it out (pictured below). But I reported this anyway since it seems to cause breakage and others might run into this.

![image](https://user-images.githubusercontent.com/537287/151659926-06433554-62db-44ad-a568-88cadf5a108b.png)

**Conclusion**
For anyone else who may encounter this, the issue can be resolved by replacing [environment]::Exit(0) with **$Form.Close()** which is the standard and expected way to close the form. [environment]::Exit(0) is a bit of a sledgehammer 🔨 and should probably not be used.
